planner/core/logical_plan_builder.go

```diff
@@ -5612,3 +5724,279 @@ func containDifferentJoinTypes(preferJoinType uint) bool {
 	}
 	return cnt > 1
 }
+
+func (b *PlanBuilder) buildCte(ctx context.Context, cte *ast.CommonTableExpression, isRecursive bool) (p LogicalPlan, err error) {
+	if isRecursive {
+		saveCheck := b.buildingRecursivePartForCTE
+		b.buildingRecursivePartForCTE = false
+		err = b.buildRecursiveCTE(ctx, cte.Query.Query)
+		if err != nil {
+			return nil, err
+		}
+		b.buildingRecursivePartForCTE = saveCheck
+	} else {
+		p, err = b.buildResultSetNode(ctx, cte.Query.Query)
+		if err != nil {
+			return nil, err
+		}
+
+		p, err = b.adjustCTEPlanSchema(p, cte)
+		if err != nil {
+			return nil, err
+		}
+
+		cInfo := b.outerCTEs[len(b.outerCTEs)-1]
+		cInfo.seedLP = p
+	}
+	return nil, nil
+}
+
+func (b *PlanBuilder) buildRecursiveCTE(ctx context.Context, cte ast.ResultSetNode) error {
```

Comment:
Specify the pattern of what recursive CTE would like. What's the default branch of the top `SWITCH` clause mean?